jerry-core/include/jerryscript-core.h

```diff
@@ -165,36 +165,31 @@ typedef uint32_t jerry_length_t;
 typedef uint32_t jerry_value_t;
 
 /**
- * Description of ECMA property descriptor.
+ * Status flags of ECMA property descriptor.
  */
-typedef struct
+typedef enum
 {
-  /** Is [[Value]] defined? */
-  bool is_value_defined;
-
-  /** Is [[Get]] defined? */
-  bool is_get_defined;
-
-  /** Is [[Set]] defined? */
-  bool is_set_defined;
-
-  /** Is [[Writable]] defined? */
-  bool is_writable_defined;
-
-  /** [[Writable]] */
-  bool is_writable;
+  JERRY_PROP_NO_OPTS = (0), /** empty property descriptor */
+  JERRY_PROP_IS_GET_DEFINED = (1 << 0), /** Is [[Get]] defined? */
+  JERRY_PROP_IS_SET_DEFINED = (1 << 1), /** Is [[Set]] defined? */
 
-  /** Is [[Enumerable]] defined? */
-  bool is_enumerable_defined;
+  JERRY_PROP_IS_CONFIGURABLE = (1 << 2), /** [[Configurable]] */
+  JERRY_PROP_IS_ENUMERABLE = (1 << 3), /** [[Enumerable]] */
+  JERRY_PROP_IS_WRITABLE = (1 << 4), /** [[Writable]] */
 
-  /** [[Enumerable]] */
-  bool is_enumerable;
+  JERRY_PROP_IS_VALUE_DEFINED = (1 << 6), /** Is [[Value]] defined? */
```

Comment:
(1 << 5) is missing

Reply:
It's missing so that we keep alignment with `ecma_property_descriptor_status_flags_t`, where (1 << 5) is the `ECMA_PROP_IS_THROW` flag, which we don't need to publish in `jerry_property_descriptor_status_flags_t`.

Reply:
I was worried about that. You cannot just "copy" the flags, since invalid flag combinations can be set by users.

Reply:
That's true. Maybe we could check that the (1 << 5) flag is not set when converting `jerry_property_descriptor_t` to `ecma_property_descriptor_t` and vice versa. The other solution is to not skip the (1 << 5) value in `jerry_property_descriptor_t`, but then we would need to copy the flags in two parts, the part before (1 << 5) and after (1 << 5), instead of a single assignment like we currently can. Which would you prefer?

Reply:
I'd prefer moving `ECMA_PROP_IS_THROW` to the end of the enum internally. For the first check I see no any special usecase why it cannot be moved, so please try it.

Reply:
I just run into this a few weeks ago. I think it must not clash with the writable/enumerable/configurable bits of the 1 byte property descriptor. So it cannot be moved freely.

Reply:
Okay, seems reasonable. Or how about publishing this flags though? TBH I dont see any disavantage of having this flags and the standard also defines `defineProperty` and `definePropetyOrThrow` operations as well. So by that we can extend the functionality of the API function.

Reply:
Maybe we could move the writable/enumerable/configurable to the first 3 bits of the prop descriptor, and the other bits in any order. Then the order of the bits woudn't be so random.

Reply:
A little update on this: I have applied your suggestions, but currently it is causing some tests to fail, I will update the PR once I fix them.